### PR TITLE
feat: onboard facebook_conversions destination

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## Description of the change

Onboard new destination Facebook Conversions.
Resolves INT-633

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by shrouti1507 — COMMENTED** [review]
please solve the sonar issue. It is asking to reduce duplication of code.

File: src/v0/destinations/facebook_conversions/config.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  'Contact',
+  'CustomizeProduct',
+  'Donate',
+  'FindLocation',
+  'Schedule',
+  'StartTrial',
+  'SubmitApplication',
+  'Subscribe',
+];
+
```
Comment (Gauravudia) [inline src/v0/destinations/facebook_conversions/config.js:91]: Why we are considering AddToWishlist and AddPaymentInfo as other standard events?
These events are supported as standard ecomm events as per our general ecomm doc
https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/#wishlist
https://www.rudderstack.com/docs/event-spec/ecommerce-events-spec/#ordering

Comment (sandeepdsvs) [reply src/v0/destinations/facebook_conversions/config.js]: updated

File: src/v0/destinations/facebook_conversions/transform.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    limitedDataUSage,
+    testDestination,
+    testEventCode,
+    datasetId,
+    accessToken,
+    actionSource,
+  } = Config;
+  const integrationsObj = getIntegrationsObj(message, DESTINATION.toLowerCase());
+
+  const endpoint = `https://graph.facebook.com/v17.0/${datasetId}/events?access_token=${accessToken}`;
```
Comment (Gauravudia) [inline src/v0/destinations/facebook_conversions/transform.js:57]: Can you move it to config?

File: src/v0/destinations/facebook_conversions/utils.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        ),
+        content_category: getContentCategory(contentCategory),
+      };
+
+      const query = eventTypeCustomData.search_string;
+      const validQueryType = ['string', 'number', 'boolean'];
+      if (query && !validQueryType.includes(typeof query)) {
+        throw new InstrumentationError("'query' should be in string format only");
+      }
+
```
Comment (Gauravudia) [inline src/v0/destinations/facebook_conversions/utils.js:135]: It is related to product searched event only, So can you move it to some function related to with name something like validatProductSearched ?

Comment (sandeepdsvs) [reply src/v0/destinations/facebook_conversions/utils.js]: Done

File: src/v0/destinations/facebook_conversions/transform.js (hunk 2 above)
Comment (sandeepdsvs) [reply src/v0/destinations/facebook_conversions/transform.js]: Done

**Review by Gauravudia — DISMISSED** [review]

**Review by Gauravudia — APPROVED** [review]

**Pull request merged**